homework_4/bets_api/internal/infrastructure/sqlite/betRepository.go

```diff
@@ -0,0 +1,255 @@
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryInsertBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "bet repository failed to insert a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryInsertBet(ctx context.Context, bet storagemodels.Bet) error {
+	// If payout is 0, do not insert it.
+	if bet.Payout == 0 {
+		insertBetSQL := "INSERT INTO bets(id, customer_id, status, selection_id, selection_coefficient, payment) VALUES (?, ?, ?, ?, ?, ?)"
+		statement, err := r.dbExecutor.PrepareContext(ctx, insertBetSQL)
+		if err != nil {
+			return err
+		}
+
+		_, err = statement.ExecContext(ctx, bet.Id, bet.CustomerId, bet.Status, bet.SelectionId, bet.SelectionCoefficient, bet.Payment)
+		return err
+	}
+
+	insertBetSQL := "INSERT INTO bets(id, customer_id, status, selection_id, selection_coefficient, payment, payout) VALUES (?, ?, ?, ?, ?, ?, ?)"
+	statement, err := r.dbExecutor.PrepareContext(ctx, insertBetSQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = statement.ExecContext(ctx, bet.Id, bet.CustomerId, bet.Status, bet.SelectionId, bet.SelectionCoefficient, bet.Payment, bet.Payout)
+	return err
+}
+
+// UpdateBet updates the provided bet in the database. An error is returned if the operation
+// has failed.
+func (r *BetRepository) UpdateBet(ctx context.Context, bet domainmodels.Bet) error {
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryUpdateBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "bet repository failed to update a bet with id "+bet.Id)
+	}
+	return nil
+}
+
+func (r *BetRepository) queryUpdateBet(ctx context.Context, bet storagemodels.Bet) error {
+	updateBetSQL := "UPDATE bets SET customer_id=?, status=?, selection_id=?, selection_coefficient=?, payment=?, payout=? WHERE id=?"
+
+	statement, err := r.dbExecutor.PrepareContext(ctx, updateBetSQL)
+	if err != nil {
+		return err
+	}
+
+	_, err = statement.ExecContext(ctx, bet.CustomerId, bet.Status, bet.SelectionId, bet.SelectionCoefficient, bet.Payment, bet.Payout, bet.Id)
+	return err
+}
+
+// GetBetByID fetches a bet from the database and returns it. The second returned value indicates
+// whether the bet exists in DB. If the bet does not exist, an error will not be returned.
+func (r *BetRepository) GetBetByID(ctx context.Context, id string) (domainmodels.BetReduced, bool, error) {
+	storageBet, err := r.queryGetBetByID(ctx, id)
+	if err == sql.ErrNoRows {
+		return domainmodels.BetReduced{}, false, nil
+	}
+	if err != nil {
+		return domainmodels.BetReduced{}, false, errors.Wrap(err, "bet repository failed to get a bet with id: "+id)
+	}
+
+	domainBet := r.betMapper.MapStorageBetToDomainBetReduced(storageBet)
+	return domainBet, true, nil
+}
+
+func (r *BetRepository) queryGetBetByID(ctx context.Context, id string) (storagemodels.Bet, error) {
+	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id='"+id+"';")
```

Comment:
You still have a couple of these where you're not using placeholders+arguments in the queries.

```suggestion
	row, err := r.dbExecutor.QueryContext(ctx, "SELECT * FROM bets WHERE id=?;", id)
```